### DISCUSSION TITLE
How do I use a pre-populated sqlite db on flutter web with drift?

### DISCUSSION THREAD

**Discussion by ehtiqamar** [simolus3/drift#2870] [Q&A] [answered]
I've added support to open a pre-populate db on native side with drift using the code provided in the docs, but I don't see any way of doing it on web with drift? Is it supported yet? If yes, please let me know how

**Comment by simolus3** [comment]
Sure, you can use the `initializeDatabase` parameter on [`WasmDatabase.open`](https://drift.simonbinder.eu/api/drift.wasm/wasmdatabase/open) to provide the database file as bytes. It's kind of documented with [this example](https://drift.simonbinder.eu/web/#migrating-from-custom-wasmdatabases) here although that is doing a lot more. But using the callback to load an asset should work fine.

**Reply by ehtiqamar** [reply]
Is the initializeDatabase method only called once i.e. when the web app is opened for the first time or is it opened everytime the web app is opened?  Also where do I load the asset to on web? Don't have much knowledge on flutter web, so any sample code would be appreciated. The one give in the reference does not seem to work

**Reply by ehtiqamar** [reply, marked as answer]
I've used this code to load my existing db on web: 

```dart
DatabaseConnection connect() {
  return DatabaseConnection.delayed(Future(() async {
    final db = await WasmDatabase.open(
      databaseName: 'sqlite',
      sqlite3Uri: Uri.parse('sqlite3.wasm'),
      driftWorkerUri: Uri.parse('drift_worker.js'),
      initializeDatabase: () async {
        logger.d("initializeDatabase: Initializing web db");

        Uint8List? oldDatabase;

        final blob = await rootBundle.load('assets/sqlite.db');
        final buffer = blob.buffer;
        oldDatabase =
            buffer.asUint8List(blob.offsetInBytes, blob.lengthInBytes);
        logger.d("Old Database: $oldDatabase");
        return oldDatabase;
      },
    );

    if (db.missingFeatures.isNotEmpty) {
      debugPrint('Using ${db.chosenImplementation} due to unsupported '
          'browser features: ${db.missingFeatures}');
    }

    return db.resolvedExecutor;
  }));
}
```

**Reply by simolus3** [reply]
Does that code not work? It looks correct to me.

> Is the initializeDatabase method only called once i.e. when the web app is opened for the first time or is it opened everytime the web app is opened?

It will only be called the first time we try to open the database.

**Reply by ehtiqamar** [reply]
You mean the code that I mentioned above or the example that you've mentioned?

**Reply by ehtiqamar** [reply]
Yea, my code did work but I'm not really sure if it is the correct approach since I've directly loaded my db from assets without checking anything where as the code in example had some checks in place for db.

**Reply by simolus3** [reply]
It is correct, I'll update the website to include this as well.
